sqlserver dialect generates invalid limit syntax for prepared statement

**Describe the bug**
SQL Server requires `SELECT TOP (@p1) ...` when generating a parameterized statement (parentheses around the parameter, only for TOP).  This library generates `SELECT TOP @p1 ...` which results in an error: mssql: Incorrect syntax near '@p1'

**To Reproduce**
I wrote the following (a copy of the TestLimitOffset test) that can be dropped right in sqlserver_test.go:

```
func (mt *sqlserverTest) TestLimitOffsetParameterized() {
	ds := mt.db.From("entry").Prepared(true).Where(goqu.C("id").Gte(1)).Limit(1)
	s, _, _ := ds.ToSQL()
	println(s)

	var e entry
	found, err := ds.ScanStruct(&e)
	mt.NoError(err)
	mt.True(found)
	mt.Equal(uint32(1), e.ID)

	ds = mt.db.From("entry").Where(goqu.C("id").Gte(1)).Order(goqu.C("id").Desc()).Limit(1)
	found, err = ds.ScanStruct(&e)
	mt.NoError(err)
	mt.True(found)
	mt.Equal(uint32(10), e.ID)

	ds = mt.db.From("entry").Where(goqu.C("id").Gte(1)).Order(goqu.C("id").Asc()).Offset(1).Limit(1)
	found, err = ds.ScanStruct(&e)
	mt.NoError(err)
	mt.True(found)
	mt.Equal(uint32(2), e.ID)
}
```

**Expected behavior**
Running the tests produces the following output:
```
SELECT  TOP @p1 * FROM "entry" WHERE ("id" >= @p2)
--- FAIL: TestSqlServerSuite (0.34s)
    --- FAIL: TestSqlServerSuite/TestLimitOffsetParameterized (0.02s)
        sqlserver_test.go:368:
            	Error Trace:	sqlserver_test.go:368
            	Error:      	Received unexpected error:
            	            	mssql: Incorrect syntax near '@p1'.
            	Test:       	TestSqlServerSuite/TestLimitOffsetParameterized
        sqlserver_test.go:369:
            	Error Trace:	sqlserver_test.go:369
            	Error:      	Should be true
            	Test:       	TestSqlServerSuite/TestLimitOffsetParameterized
        sqlserver_test.go:370:
            	Error Trace:	sqlserver_test.go:370
            	Error:      	Not equal:
            	            	expected: 0x1
            	            	actual  : 0x0
            	Test:       	TestSqlServerSuite/TestLimitOffsetParameterized
FAIL
exit status 1
FAIL	github.com/doug-martin/goqu/v9/dialect/sqlserver	0.465s
```

**Dialect:**
- [X] sqlserver

**Additional context**
This is a known SQL Server limitation that other people have also experienced in DB drivers, example: https://stackoverflow.com/questions/56132497/parametrizing-top-value-in-tsql-and-pyodbc

I looked into common_sql_generator.go, LimitSQL, but since it only takes a fragment, not a template string or similar, there's no way to wrap the limit parameter.  That might have to be an additional piece of configuration in the dialect, but I suspect you might have something similar used elsewhere that would be more appropriate/well-designed than my theoretical hack of a template string.

--- a/dialect/sqlserver/sqlserver.go
+++ b/dialect/sqlserver/sqlserver.go
@@ -23,6 +23,7 @@ func DialectOptions() *goqu.SQLDialectOptions {
 	opts.SupportsWithCTERecursive = false
 	opts.SupportsDistinctOn = false
 	opts.SupportsWindowFunction = false
+	opts.SurroundLimitWithParentheses = true
 
 	opts.PlaceHolderFragment = []byte("@p")
 	opts.LimitFragment = []byte(" TOP ")

--- a/dialect/sqlserver/sqlserver_test.go
+++ b/dialect/sqlserver/sqlserver_test.go
@@ -358,6 +358,27 @@ func (mt *sqlserverTest) TestLimitOffset() {
 	mt.Equal(uint32(2), e.ID)
 }
 
+func (mt *sqlserverTest) TestLimitOffsetParameterized() {
+	ds := mt.db.From("entry").Prepared(true).Where(goqu.C("id").Gte(1)).Limit(1)
+	var e entry
+	found, err := ds.ScanStruct(&e)
+	mt.NoError(err)
+	mt.True(found)
+	mt.Equal(uint32(1), e.ID)
+
+	ds = mt.db.From("entry").Prepared(true).Where(goqu.C("id").Gte(1)).Order(goqu.C("id").Desc()).Limit(1)
+	found, err = ds.ScanStruct(&e)
+	mt.NoError(err)
+	mt.True(found)
+	mt.Equal(uint32(10), e.ID)
+
+	ds = mt.db.From("entry").Prepared(true).Where(goqu.C("id").Gte(1)).Order(goqu.C("id").Asc()).Offset(1).Limit(1)
+	found, err = ds.ScanStruct(&e)
+	mt.NoError(err)
+	mt.True(found)
+	mt.Equal(uint32(2), e.ID)
+}
+
 func (mt *sqlserverTest) TestInsert() {
 	ds := mt.db.From("entry")
 	now := time.Now()

--- a/sqlgen/common_sql_generator.go
+++ b/sqlgen/common_sql_generator.go
@@ -104,7 +104,13 @@ func (csg *commonSQLGenerator) OrderWithOffsetFetchSQL(
 func (csg *commonSQLGenerator) LimitSQL(b sb.SQLBuilder, limit interface{}) {
 	if limit != nil {
 		b.Write(csg.dialectOptions.LimitFragment)
+		if csg.dialectOptions.SurroundLimitWithParentheses {
+			b.WriteRunes(csg.dialectOptions.LeftParenRune)
+		}
 		csg.esg.Generate(b, limit)
+		if csg.dialectOptions.SurroundLimitWithParentheses {
+			b.WriteRunes(csg.dialectOptions.RightParenRune)
+		}
 	}
 }
 

--- a/sqlgen/sql_dialect_options.go
+++ b/sqlgen/sql_dialect_options.go
@@ -45,6 +45,9 @@ type (
 		// Set to true if the dialect requires join tables in UPDATE to be in a FROM clause (DEFAULT=true).
 		UseFromClauseForMultipleUpdateTables bool
 
+		// Surround LIMIT parameter with parentheses, like in MSSQL: SELECT TOP (10) ...
+		SurroundLimitWithParentheses bool
+
 		// The UPDATE fragment to use when generating sql. (DEFAULT=[]byte("UPDATE"))
 		UpdateClause []byte
 		// The INSERT fragment to use when generating sql. (DEFAULT=[]byte("INSERT INTO"))

